### ISSUE TITLE
feat: make system config available as separate file

### ISSUE THREAD

**Issue by jeffbcross** [angular/angular-cli#429]
I'm creating an [App Shell](https://developers.google.com/web/updates/2015/11/app-shell?hl=en) broccoli plugin that needs to access a project's `System` config to properly load dependencies of an app component when pre-compiling. Right now, I would have to parse `index.html` to find the system configuration. Ideally, there would be a file I could import instead, i.e. `system.config.json`.

**Comment by jeffbcross** [comment]
This also brings up another challenge, which is that my plugin doesn't know how to locate libraries being loaded via script tags instead of by `System`. I would have to manually augment the `System` configuration in my bundle to include angular, http, router, and rxjs.

**Comment by Brocco** [comment]
Were you looking for more than just this:
**index.html**
```
<script src="system-config.js"></script>
```
**system-config.ts**
```
declare var System: any;

System.config({
  packages: {
    app: {
      format: 'register',
      defaultExtension: 'js'
    }
  }
});
System.import('main.js').then(null, console.error.bind(console));
```

**Comment by jeffbcross** [comment]
@Brocco in my case, that would work (except from the `System.import` included at the end, which would cause problems). It would be even better if the config could be a JSON file, so add-ons could inspect it and modify it before calling `System.config`

**Comment by Brocco** [comment]
FYI, I have the separation into a separate file implemented in the style guide PR #427 but would be ideal if someone could go the extra step and get that into a JSON file

**Comment by Brocco** [comment]
Resolved via #427
